File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Action/Animations/TalkAction.ts
import Async from "../../../../../Core/Async";
import DungeonManager from "../../DungeonManager";
import { SpeechBubble } from "../../Effect/Combat/SpeechBubble";
import AbstractGameAction, { ActionType } from "../AbstractGameAction";

export class TalkAction extends AbstractGameAction{
    Type = ActionType.TEXT;
    public constructor(private X:number,private Y:number,private Text:string,private FlipX? : boolean,private Interval:number = 2,private BubbleDuration:number = 2) {
        super();
    }
    
    public async Update(): Promise<boolean> {
        DungeonManager.EffectManager.Play(new SpeechBubble(this.X,this.Y,this.Text,this.BubbleDuration,this.FlipX));
        await Async.Delay(this.Interval);
        return true;
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Action/Combat/TalkAction.cs
using Cysharp.Threading.Tasks;
using ModTheSpire.Dungeon;
using ModTheSpire.Effect.Combat;
using ModTheSpire.Event;

namespace ModTheSpire.Effect.Other
{
    public class TalkAction : IGameAction
    {
        private readonly float mX;
        private readonly float mY;
        private readonly string mText;
        private readonly bool mFlipX;
        private readonly int mInterval;
        private readonly int mBubbleDuration;
        public float Duration { get; set; }
        public ActionType Type { get; } = ActionType.TEXT;
        
        public TalkAction(float x,float y,string text,bool flipX = false,int interval = 2000,int bubbleDuration = 2000)
        {
            mX = x;
            mY = y;
            mText = text;
            mFlipX = flipX;
            mInterval = interval;
            mBubbleDuration = bubbleDuration;
        }
        public void Init()
        {
            throw new System.NotImplementedException();
        }

        public async UniTask<bool> Update()
        {
            DungeonManager.EffectManager.Play(new SpeechBubble(this.mX,this.mY,this.mText,this.mBubbleDuration,this.mFlipX));
            await UniTask.Delay(this.mInterval);
            return true;
        }
    }
}